### PR TITLE
Time Traveler (set 1) - promoted to working

### PR DESCRIPTION
**Pull request by MooglyGuy**
Machines promoted to working
----------------------------
Time Traveler (set 1) [Ryan Holtz, Matt O, ld-decode Team]

-laserdsc: Added functionality to retrieve if a disc is CAV. [Ryan Holtz]

-ldv4200hle: HLE Pioneer LD-V4200 player. [Ryan Holtz]

### CODE REVIEW

File: src/devices/machine/ldv4200hle.h [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+// device type definition
+DECLARE_DEVICE_TYPE(PIONEER_LDV4200HLE, pioneer_ldv4200hle_device)
+
+
+
+//**************************************************************************
+//  GLOBAL HELPERS
+//**************************************************************************
+
+static constexpr uint16_t make_ldv4000_command(const char *str) { return ((uint8_t)str[0] << 8) | (uint8_t)str[1]; }
```
Comment (cuavas) [inline src/devices/machine/ldv4200hle.h:33]: Can you put this inside the class?  It should still work without being in the global namespace.

File: src/devices/machine/ldv4200hle.h [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+	void add_command_byte(uint8_t data);
+
+	void queue_reply(const char *reply);
+	void queue_error(error_code err);
+
+	void normalize_command_buffer();
+	void process_command_buffer();
+
+	uint32_t bcd_to_literal(uint32_t bcd);
+	bool is_number(char value);
```
Comment (cuavas) [inline src/devices/machine/ldv4200hle.h:170]: It should be possible to make both of these `static` member functions.

File: src/mame/drivers/timetrv.cpp [hunk 3]
```diff
@@ -119,60 +225,106 @@ static INPUT_PORTS_START( timetrv )
 	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_COIN2 )
```
Comment (cuavas) [inline src/mame/drivers/timetrv.cpp:225]: If this is for a bill acceptor/banknote reader, you can use `IPT_BILL1`.

File: src/mame/layout/timetrv.lay [hunk 4]
```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0"?>
+<!--
+license:BSD-3-Clause
```
Comment (cuavas) [inline src/mame/layout/timetrv.lay:3]: Is there a specific reason for a BSD-licensed layout?  We generally use CC0 for internal layouts to reduce the hassle for people using them as examples to follow.  If there’s a reason for it, it’s fine though.

**Review by cuavas — COMMENTED** [review]
Looks good.  Just a few minor thing and we should be able to have this in.

File: src/mame/layout/timetrv.lay (hunk 4 above)
Comment (MooglyGuy) [reply src/mame/layout/timetrv.lay]: I did notice that all of the internal layouts I had looked at were CC0, but I don't recall seeing the justification for it stated until now. Completely understandable, and I've adjusted the license to be CC0.

File: src/mame/drivers/timetrv.cpp (hunk 3 above)
Comment (MooglyGuy) [reply src/mame/drivers/timetrv.cpp:226]: Time Traveler had two coin slots and a separate bill validator. The hookup for the bill validator is not yet known, I've now added a note to that effect at the top of the driver.

File: src/devices/machine/ldv4200hle.h (hunk 2 above)
Comment (MooglyGuy) [reply src/devices/machine/ldv4200hle.h]: Done, nice catch.

File: src/devices/machine/ldv4200hle.h (hunk 1 above)
Comment (MooglyGuy) [reply src/devices/machine/ldv4200hle.h:35]: I did, but it resulted in a "[make_ldv4000_command] called in a constant expression before its definition is complete" error for each entry in the enum list. I would have called it "make_command" were it in the class definition itself, but I added the "ldv4000" qualifier to at least reduce the chance of a global namespace collision. I've added a comment clarifying to this effect.

File: src/devices/machine/laserdsc.h [hunk 5]
```diff
@@ -233,6 +233,7 @@ class laserdisc_device :    public device_t,
 	int32_t generic_update(const vbi_metadata &vbi, int fieldnum, const attotime &curtime, player_state_info &curstate);
 
 	// general helpers
+	bool is_cav_disc() { return m_is_cav_disc; }
```
[the remark this answers is not in the document]
Comment (MooglyGuy) [reply src/devices/machine/laserdsc.h]: Nice catch. A lot more stuff in that file can be `const`-ified, but it's better to at least not add to the problem. Fixed. :)

**Pull request merged**